### PR TITLE
docs: add AGENTS.md with project code standards

### PR DESCRIPTION
**Pull request by rmartz**
Adds `AGENTS.md` containing code standards distilled from the sibling projects (firebase-nextjs-template, hidden-role-game, personal-budget, trip-split, group-picks). Only conventions relevant to this TypeScript Node.js CLI project are included — React/Next.js, Storybook, and app-specific sections were omitted.

Sections covered: package manager (pnpm), common commands, TypeScript strict-mode rules, code conventions, file/test organization, test design principles, and git/PR conventions.

`CLAUDE.md` is a symlink to `AGENTS.md` so both Claude Code (which reads `CLAUDE.md`) and other AI tooling (which reads `AGENTS.md`) see the same content from a single source of truth.

---
*Created by Claude Sonnet 4.6*

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+## Package Manager
+
+- Always use `pnpm`. Never `npm` or `yarn`.
+
+## Common Commands
+
+```bash
+pnpm build         # Compile TypeScript to dist/
+pnpm test          # Run BATS shell tests
+pnpm run test:ts   # Run Vitest unit tests
```
[the remark this answers is not in the document]
Comment (rmartz) [reply AGENTS.md:12]: Dismissed — PR #35 is open and migrates the repo to pnpm (lockfile, `packageManager` field, CI). AGENTS.md documents the target state; this section will be accurate once #35 merges.

---
*Posted by Claude Sonnet 4.6*

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+## TypeScript
+
+- Strict mode throughout. No `any` types. No `@ts-ignore`.
+- Do not use `null` unless required for API compatibility or when explicitly distinguishing `null` from `undefined`. Prefer `undefined` for absent/optional values throughout the codebase.
```
[the remark this answers is not in the document]
Comment (rmartz) [reply AGENTS.md:18]: Dismissed — PR #43 just landed the `null → undefined` refactor for `Options.init`, so the codebase now matches this convention.

---
*Posted by Claude Sonnet 4.6*

File: AGENTS.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Code Standards
+
+## Package Manager
+
+- Always use `pnpm`. Never `npm` or `yarn`.
```
[the remark this answers is not in the document]
Comment (rmartz) [reply AGENTS.md:5]: Dismissed — `CLAUDE.md` is committed as a mode-120000 symlink pointing to `AGENTS.md`. GitHub's diff view does not render symlinks, but `git ls-tree` confirms the symlink is present in the branch. Both files resolve to the same content at runtime.

---
*Posted by Claude Sonnet 4.6*

File: AGENTS.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- **No spurious variables.** Do not assign a value to a variable only to immediately return it on the next line — return the expression directly instead.
+- **No IIFEs.** Do not use immediately-invoked function expressions. Extract the logic into a named helper function or compute the value with a plain expression instead.
+- **No function-style imports.** Do not use inline `import("…").Type` syntax in type annotations. Use module-level `import type { … } from "…"` statements at the top of the file.
+- **No unnecessary helpers.** Do not extract logic into a helper function unless it separates significant logic or belongs in a different module. Three similar lines is better than a premature abstraction.
+- **Enums and constant objects** should be kept in alphabetical order to minimize merge conflicts.
+
+## File Organization
+
+- **Source files**: Keep under ~200 lines (split at ~240). Large files should be split by logical concern.
+- **Test files**: Keep under ~300 lines (split at ~360). Place in `src/__tests__/` using `.test.ts` extension. When splitting, organize into a `{module}-tests/` subdirectory.
```
[the remark this answers is not in the document]
Comment (rmartz) [reply AGENTS.md]: Fixed in 7529399 — softened to a target rather than a hard limit, and explicitly named `rotate-keys.ts` as a known exception until it is refactored.

---
*Posted by Claude Sonnet 4.6*

File: AGENTS.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+- **Granularity scales with level of abstraction.** Low-level functions (pure utilities, parsers) warrant thorough edge-case coverage. High-level functions (service orchestration) should have smoke tests that verify they correctly apply the lower-level logic — not re-test every edge case that belongs in the lower-level tests.
+
+## Git Conventions
+
+- Branch names: lowercase with hyphens, prefixed by type: `feat/`, `fix/`, `chore/`, `refactor/`, `docs/`, `ci/` (e.g., `feat/yaml-driven-init-config`).
+- Commit messages: imperative verbs (Add, Implement, Fix, Update, Extract, Remove). No `feat:`/`fix:` prefixes within a feature branch.
+- PR titles must follow Conventional Commits format: `<type>: description` or `<type>(<scope>): description`. Valid types: `feat`, `fix`, `docs`, `chore`, `refactor`, `test`, `style`, `perf`, `ci`, `build`, `revert`. A `!` suffix is allowed before the colon to denote breaking changes (e.g., `feat!: remove legacy auth`). This is enforced by CI.
```
[the remark this answers is not in the document]
Comment (rmartz) [reply AGENTS.md]: Fixed in 7529399 — added a note that this repo uses squash merges, so only the PR title (Conventional Commits) enters main's history and drives semantic-release versioning. Feature-branch commits are intentionally non-CC.

---
*Posted by Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

All 5 Copilot threads addressed:

- ✅ [AGENTS.md line 12 — pnpm vs npm tooling](https://github.com/rmartz/vercel-deploy-scripts/pull/36#discussion_r3181790132) — Dismissed: moot once PR #35 (pnpm migration) merges; AGENTS.md documents the target state.
- ✅ [AGENTS.md line 18 — null vs undefined convention](https://github.com/rmartz/vercel-deploy-scripts/pull/36#discussion_r3181790180) — Dismissed: PR #43 just landed the `null → undefined` refactor, so codebase now matches.
- ✅ [AGENTS.md line 33 — file size limits](https://github.com/rmartz/vercel-deploy-scripts/pull/36#discussion_r3181790204) — Fixed in 7529399: softened to a target, named `rotate-keys.ts` as a known exception.
- ✅ [AGENTS.md line 52 — commit message guidance vs semantic-release](https://github.com/rmartz/vercel-deploy-scripts/pull/36#discussion_r3181790232) — Fixed in 7529399: clarified that squash merges mean only the PR title (Conventional Commits) enters main's history.
- ✅ [AGENTS.md line 5 — CLAUDE.md symlink not in diff](https://github.com/rmartz/vercel-deploy-scripts/pull/36#discussion_r3181790262) — Dismissed: symlink is committed as mode 120000; GitHub's diff view simply doesn't render symlinks.

## Verdict

All prior items resolved, no new issues — ready to merge.

---
*Claude Sonnet 4.6*
